ZukoAzula2016/src/org/team1540/zukoazula/Shooter.java

```diff
@@ -30,44 +32,101 @@
-        BooleanInput warmup = ZukoAzula.controlBinding.addBoolean("Shooter Warm-Up");
-        BooleanInput fire = ZukoAzula.controlBinding.addBoolean("Shooter Fire");
-        BooleanInput spinup = warmup.or(fire).or(actuallyFiring);
-
-        EventInput intakeArmRollerForward = ZukoAzula.controlBinding.addEvent("Intake Arm Rollers Forward");
-        EventInput intakeArmRollerBackward = ZukoAzula.controlBinding.addEvent("Intake Arm Rollers Backward");
-        EventInput intakeArmRollerStop = ZukoAzula.controlBinding.addEvent("Intake Arm Rollers Stop");
-
-        StateMachine states = new StateMachine(0, "passive", "forward", "backward");
-        states.setStateWhen("passive", intakeArmRollerStop.or(spinup.onPress()).or(FRC.startTele));
-        states.setStateWhen("forward", intakeArmRollerForward.andNot(spinup));
-        states.setStateWhen("backward", intakeArmRollerBackward.andNot(spinup));
-
-        states.getIsState("passive").toFloat(states.getIsState("forward").toFloat(-1, 1), 0).send(intakeArmRollerCAN.simpleControl());
-
-        FloatInput indexerIntakeSpeed = ZukoAzula.mainTuning.getFloat("Indexer Speed During Intake", 1f);
-        FloatInput indexerPassiveSpeed = ZukoAzula.mainTuning.getFloat("Roller Passive Speed", 0.1f);
-
-        rollerSpeed.attach(rollerArb.addBehavior("Not Shooting", FRC.inTeleopMode().and(shooter.isStopped)),
-                states.getIsState("passive").toFloat(indexerIntakeSpeed.negatedIf(states.getIsState("backward")), indexerPassiveSpeed));
-
-        BooleanInput shouldSpinUp = setupRollersForSpinup(spinup, actuallyFiring);
-
-        shooter.setup(shouldSpinUp);
-
-        fire.and(shooter.isUpToSpeed).onPress().send(actuallyFiring);
+        
+        StateMachine flywheelTarget = new StateMachine(0, "off", "low", "high");
+        StateMachine flywheelActual = new StateMachine(0, "off", "low", "high");
+        
+        StateMachine shooterStates = new StateMachine(0, 
+                "passive", // do nothing
+                "ejecting", // when the ball is being ejected from the intake mechanism
+                "intaking", // when the intake mechanism is intaking a ball
+                "loaded", // when a ball is loaded in the intake mechanism
+                "cocking", // when a ball is being cocked so that it is not in contact w/ the flywheel 
+                "spinup",  // when the flywheel is spinning up to maximum speed
+                "firing"); // when the ball is firing
+        
+        TalonExtendedMotor flywheel = makeLinkedTalons();
+        flywheel.modEncoder().configureEncoderCodesPerRev(125*15);
+        flywheel.modGeneralConfig().getBrakeNotCoast().set(false);
+        FloatInput flywheelSpeed = flywheel.modEncoder().getEncoderVelocity().absolute();
+        FloatOutput flywheelSimple = flywheel.simpleControl().addRamping(ZukoAzula.mainTuning.getFloat("Shooter Flywheel Ramping", 0.01f).get(), 
+                FRC.constantPeriodic);
```

Comment:
So... that's not really how tuning works. Unfortunately I don't have much of a better option, but it might just be better to hardcode it if you're going to do it like that.